### ISSUE TITLE
Exception 'The limit of '0' for Top query has been exceeded' by calling examples with parameter "top"

### ISSUE THREAD

**Issue by dmvlasenk** [dotnet/aspnet-api-versioning#944]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

BUG: 
![image](https://user-images.githubusercontent.com/26723311/208975238-2ff65bfe-0976-4793-824f-1862c0f642cf.png)

Step to reproduce: 
- checkout branch  "release/6.3"
- Execute \examples\AspNetCore\OData\ODataOpenApiExample\
- Call https://localhost:5001/api/Orders?api-version=2.0&top=1

I assume that the parameter "top" generates exceptions for all OData examples having `options.Validate` or [EnableQuery]. 

### Expected Behavior

Response 200

### Exceptions (if any)

Microsoft.OData.ODataException: 'The limit of '0' for Top query has been exceeded. The value from the incoming request is '1'.'

### .NET Version

net6.0

**Comment by commonsensesoftware** [comment]
This is ultimately an issue with OData; however, I didn't want to just punt you over to their repo. I've updated any existing issue with comprehensive details regarding the problem in OData/AspNetCoreOData#695.

TL;DR

The crux of the problem is that `Order` has model bound settings applied a result of `[Select]` being used. This cause all other `MaxTop` settings to be ignored.  One fix is to apply `[Page(MaxTop = 42)]`. That would apply to all API versions, but that might be ok. The other way is to apply it via the model builder:

```c#
builder.EntitySet<Order>("Orders").EntityType.Page(maxTopValue: 42, default);
```

This can easily be applied via `IModelConfiguration` (e.g. **OrderConfiguration.cs**) and can vary by API version. Model bound settings are enforced before `[EnableQuery]` so those settings end up being ignored.

It was pretty tedious to track down the main cause. I'll update the examples so that they at least work as expected.
